iac/stacks/common.py

```diff
@@ -0,0 +1,140 @@
+from aws_cdk import (
+    Duration,
+    RemovalPolicy,
+    Stack,
+    aws_ec2 as ec2,
+    aws_rds as rds,
+    aws_s3 as s3,
+    aws_ecs as ecs,
+    aws_elasticloadbalancingv2 as elb,
+)
+from constructs import Construct
+
+from iac.settings import ProjectSettings
+
+
+class CommonStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        config: ProjectSettings,
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        self.vpc = ec2.Vpc(
+            self,
+            "Vpc",
+            cidr="10.10.0.0/16",
+            max_azs=3,
+            nat_gateways=1,
+            subnet_configuration=[
+                ec2.SubnetConfiguration(
+                    subnet_type=ec2.SubnetType.PUBLIC,
+                    name="Ingress",
+                    map_public_ip_on_launch=True,
+                    cidr_mask=24,
+                ),
+                ec2.SubnetConfiguration(
+                    subnet_type=ec2.SubnetType.PRIVATE_WITH_NAT,
+                    name="Application",
+                    cidr_mask=20,
+                ),
+                ec2.SubnetConfiguration(
+                    subnet_type=ec2.SubnetType.PRIVATE_ISOLATED,
+                    name="Database",
+                    cidr_mask=24,
+                ),
+            ],
+        )
+
+        self.database = rds.DatabaseInstance(
+            self,
+            "PostgresRds",
+            vpc=self.vpc,
+            engine=rds.DatabaseInstanceEngine.POSTGRES,
+            instance_type=ec2.InstanceType.of(
+                ec2.InstanceClass.BURSTABLE3, ec2.InstanceSize.MICRO
+            ),
+            # database_name=config.database.name,
+            vpc_subnets=ec2.SubnetSelection(
+                subnet_type=ec2.SubnetType.PRIVATE_ISOLATED
+            ),
+            backup_retention=Duration.days(7),
+            deletion_protection=True,
+            removal_policy=RemovalPolicy.SNAPSHOT,
+        )
+
+        bastion = ec2.BastionHostLinux(
```

Comment:
Didn't we decide we don't need a dedicated bastion, if I can ssh into a fargate container and tunnel to postgres from there?

Reply:
Let's discuss this further... outside of this ticket. We already have 2 tickets on this topic: 1702 and 1698

Reply:
I would warn against getting in the habit of jumping on a prod container. Perhaps a bastion host is good to have. It's only a t3.nano ($4/mo)

Reply:
removed